server/src/views/frame.js

```diff
@@ -222,6 +222,7 @@ class Frame extends React.Component {
 
   clickFullPageButton(e) {
     e.preventDefault();
+    history.pushState({state: "frame"}, "", "#frame");
```

Comment:
If we use location.hash, won't the browser scroll right to the element? We want to scroll to the element minus a bit. I think this implementation is better.